src/main/java/orionmipt/LJV.java

```diff
@@ -0,0 +1,276 @@
+                            + "[label=\"" + name + "\",fontsize=12"
+                            + (fabs == null ? "" : "," + fabs)
+                            + "];\n");
+                    generateDotInternal(ctx, idGenerator, ref, out, visited);
+                } catch (IllegalAccessException e) {
+                }
+            }
+        }
+    }
+
+    private static boolean redefinesToString(Object obj) {
+        Method[] ms = obj.getClass().getMethods();
+        for (int i = 0; i < ms.length; i++)
+            if (ms[i].getName().equals("toString") && ms[i].getDeclaringClass() != Object.class)
+                return true;
+        return false;
+    }
+
+
+    protected String className(Object obj, Context context, boolean useToStringAsClassName) {
+        if (obj == null)
+            return "";
+
+        Class<?> c = obj.getClass();
+        if (useToStringAsClassName && redefinesToString(obj))
+            return Quote.quote(obj.toString());
+        else {
+            String name = c.getName();
+            if (!context.isShowPackageNamesInClasses() || c.getPackage() == LJV.class.getPackage()) {
+                //- Strip away everything before the last .
+                name = name.substring(name.lastIndexOf('.') + 1);
+
+                if (!context.isQualifyNestedClassNames())
+                    name = name.substring(name.lastIndexOf('$') + 1);
+            }
+            return name;
+        }
+    }
+
+    boolean canTreatAsPrimitive(Context context, Object obj) {
+        return obj == null || canTreatClassAsPrimitive(context, obj.getClass());
+    }
+
+
+    private boolean canTreatClassAsPrimitive(Context context, Class<?> cz) {
+        if (cz == null || cz.isPrimitive())
+            return true;
+
+        if (cz.isArray())
+            return false;
+
+        do {
+            if (context.isTreatsAsPrimitive(cz)
+                    || context.isTreatsAsPrimitive(cz.getPackage())
+            )
+                return true;
+
+            if (cz == Object.class)
+                return false;
+
+            Class<?>[] ifs = cz.getInterfaces();
+            for (int i = 0; i < ifs.length; i++)
+                if (canTreatClassAsPrimitive(context, ifs[i]))
+                    return true;
+
+            cz = cz.getSuperclass();
+        } while (cz != null);
+        return false;
+    }
+
+    boolean looksLikePrimitiveArray(Object obj, Context context) {
+        Class<?> c = obj.getClass();
+        if (c.getComponentType().isPrimitive())
+            return true;
+
+        for (int i = 0, len = Array.getLength(obj); i < len; i++)
+            if (!canTreatAsPrimitive(context, Array.get(obj, i)))
+                return false;
+        return true;
+    }
+
+    private void generateDotInternal(Context ctx, IdGenerator idGenerator, Object obj, StringBuilder out, Set<Object> visited)
+            throws IllegalArgumentException {
+        if (visited.add(new VisitedObject(obj))) {
```

Comment:
there's no need in wrapping `Object` into `VisitiedObject`, if we are using `IdentityHashMap`-backed set for `visited`.